src/cpp/src/lm_encoding.cpp

```diff
@@ -29,6 +29,23 @@ void update_position_ids(ov::Tensor&& position_ids, const ov::Tensor&& attention
     }
 }
 
+void update_3d_position_ids(ov::Tensor&& position_ids, const ov::Tensor&& attention_mask, const int64_t rope_delta) {
+    const size_t batch_size = attention_mask.get_shape().at(0);
+    const size_t sequence_length = attention_mask.get_shape().at(1);
+    const size_t num_dimensions = 3;
+
+    position_ids.set_shape({num_dimensions, batch_size, 1});
```

Comment:
It's true that `position_ids` has `num_dimensions` dimensions. But you use `num_dimensions` as the size of the first dimension instead of the total number of dimensions. Rename this to number of channels/colors.

Reply:
Fixed